lib/reverse_sentence.rb

```diff
@@ -1,6 +1,47 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) multiple reverses that are not nested.
+# Space complexity: O(1) constant number of variables
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return "" if my_sentence.nil?
+  my_reverse(my_sentence)
+  i = 0
+  j = 0
+  index = 0
+  set_first_letter = false
+
+  while index < my_sentence.length
+    letter = my_sentence[index]
+
+    if letter != " " && set_first_letter == false
+      i = index
+      set_first_letter = true
+    end
+
+    if set_first_letter == true && letter == " "
+      j = index - 1
+      my_sentence[i..j] = my_reverse(my_sentence[i..j])
```

Comment:
Just note that `my_sentence[i..j]` actually creates a new string, but if you assume the words are small, this is still O(1) space complexity.